projects/bird-atlas/src/app/+grid/grid-info/grid-info.component.ts

```diff
@@ -1,9 +1,172 @@
+
+const getGeoJSONFeature = (ykj: string) => {
+  const langLngStr = ykj.split(':');
+  return convertYkjToGeoJsonFeature(langLngStr[0], langLngStr[1]);
+};
+
+const getMapData = (ykj: string) => ({
+  featureCollection: {
+    features: [
+      getGeoJSONFeature(ykj),
+    ],
+    type: 'FeatureCollection'
+  },
+  getFeatureStyle: () => ({
+    weight: 5,
+    opacity: 1,
+    fillOpacity: 0.3,
+    color: '#00aa00'
+  })
+});
+
+const getAdjacentSqLink = (coords: string, latDelta: number, lngDelta: number): string[] => {
+  const c = coords.split(':');
+  const lat = parseInt(c[0], 10);
+  const lng = parseInt(c[1], 10);
+  return ['..', `${lat + latDelta}:${lng + lngDelta}`];
+};
+
+interface DatatableRow {
+  idx: number;
+  species: string;
+  code: string;
+  class: string;
+}
+
+interface GridInfoData {
+  elem: AtlasGridSquare;
+  rows: DatatableRow[];
+  status: number;
+}
 
 @Component({
   selector: 'ba-grid-info',
-  template: ``
+  templateUrl: './grid-info.component.html',
+  styleUrls: ['./grid-info.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class GridInfoComponent {
+export class GridInfoComponent implements AfterViewInit, OnDestroy {
+  @ViewChild('lajiMap', { static: false }) lajiMapElem: ElementRef;
+
+  private unsubscribe$ = new Subject<void>();
+  private loadMap$ = new BehaviorSubject<string>(undefined);
+  private map: any;
+
+  data$: Observable<GridInfoData>;
+  cols: TableColumn[];
+  loading = true;
+  datatableClasses = datatableClasses;
+  getAdjacentSqLink = getAdjacentSqLink;
+
+  constructor(
+    private zone: NgZone,
+    private atlasApi: AtlasApiService,
+    private route: ActivatedRoute,
+    private breadcrumbs: BreadcrumbService,
+    private translate: TranslateService,
+    private headerService: HeaderService,
+    private cdr: ChangeDetectorRef
+  ) {
+    this.data$ = this.route.paramMap.pipe(
+      tap(() => {
+        this.breadcrumbs.setBreadcrumbName(BreadcrumbId.GridInfo, undefined);
+        this.loading = true;
+        this.cdr.markForCheck();
+      }),
+      switchMap(params => this.atlasApi.getGridSquare(params.get('id'))),
+      map(elem => ({
+        elem,
+        rows: elem.data.map((d, idx) => ({
+          idx: idx + 1,
+          species: d.speciesName,
+          code: (<string>d.atlasCode.key).match(/[0-9]+/)?.[0] || null,
+          class: d.atlasClass.value
+        })),
+        status: 200
+      })),
+      tap(d => {
+        this.breadcrumbs.setBreadcrumbName(
+          BreadcrumbId.GridInfo,
+          d.elem.coordinates
+        );
+        this.headerService.setHeaders({
+          title: `${d.elem.name} ${d.elem.coordinates} | ${this.translate.instant('ba.header.title')}`
+        });
+        this.loadMap$.next(d.elem.coordinates);
+      }),
+      catchError(() => of({
+        elem: undefined,
+        rows: [],
+        status: 404
+      })),
+      tap(() => this.loading = false)
```

Comment:
You might need to change this to `delay(0), tap(() => this.loading = false)` (or add `cdr.detectChanges()`) since it looks like the same can happen here than in the `species-info.component` where the spinner spins indefinitely

Reply:
Looks like AtlasApi no longer returns the coordinates field. @MLumme can this field make a comeback, or perhaps the id field should be returned without the tun.fi part?

Reply:
Added the `detectChanges()`. Imo using `detectChanges()` rather than `delay(0)` feels more natural and the role of it is a perhaps a bit less mysterious.

Reply:
> Looks like AtlasApi no longer returns the coordinates field. @MLumme can this field make a comeback, or perhaps the id field should be returned without the tun.fi part?

Should be safe to merge since this is apparently a bug in the backend.

Reply:
Ok thanks! I think this is ready then